sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/MethodSignatureHelper.cs

```diff
@@ -0,0 +1,54 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace SonarAnalyzer.Helpers
+{
+    public class MethodSignatureHelper
+    {
+        public static Func<MethodSignature, bool> IsSameMethod(SimpleNameSyntax identifierName, SemanticModel semanticModel)
+        {
+            var identifierText = identifierName.Identifier.ValueText;
+            var identifierSymbol = new Lazy<IMethodSymbol>(() => semanticModel.GetSymbolInfo(identifierName).Symbol as IMethodSymbol);
+            // This function will be called multiple times for each tracked MethodSignature,
+            // hence we cache as much as possible before.
+            return (methodSignature) =>
+                identifierText == methodSignature.Name &&
+                identifierSymbol.Value != null &&
+                identifierSymbol.Value.ContainingType.Is(methodSignature.ContainingType);
+        }
+
+        public static Func<MethodSignature, bool> IsSameProperty(SimpleNameSyntax identifierName, SemanticModel semanticModel)
+        {
+            var identifierText = identifierName.Identifier.ValueText;
+            var identifierSymbol = new Lazy<IPropertySymbol>(() => semanticModel.GetSymbolInfo(identifierName).Symbol as IPropertySymbol);
+            // This function will be called multiple times for each tracked MethodSignature,
+            // hence we cache as much as possible before.
+            return (methodSignature) =>
+                identifierText == methodSignature.Name &&
+                identifierSymbol.Value != null &&
+                identifierSymbol.Value.ContainingType.Is(methodSignature.ContainingType);
```

Comment:
_IsSameProperty_ and _IsSameMethod_ are identical apart from the symbol type, which isn't visible outside the methods, and presumably there are equivalent methods for VB. Can we avoid the repetition?